openeo/rest/connection.py

```diff
@@ -588,6 +587,30 @@ def describe_collection(self, name) -> dict:
         data = self.get('/collections/{}'.format(name)).json()
         return VisualDict("collection", data = data)
 
+    def collection_items(self, name, spatial_extent: Optional[Dict[str, float]] = None, temporal_extent: Optional[List[Union[str, datetime.datetime]]] = None, limit: int = None) -> Iterator[dict]:
+        """
+        Loads items for a specific image collection.
+        May not be available for all collections.
+
+        :param name: String Id of the collection
+        :param spatial_extent: Limits the items to the given bounding box.
+        :param temporal_extent: Limits the items to the specified temporal interval.
+        :param limit: The amount of items per request/page. If None, the back-end decides.
+        The interval has to be specified as an array with exactly two elements (start, end).
+        Also supports open intervals by setting one of the boundaries to None, but never both.
+        :return: data_list: List A list of items
+        """
+        url = '/collections/{}/items'.format(name)
+        params = {}
+        if spatial_extent:
+            params["bbox"] = ",".join(str(c) for c in spatial_extent)
```

Comment:
`spatial_extent` is typed as `Dict[str, float]` (so a bounding box dictionary I guess), but `str(c) for c in spatial_extent` will iterate over the  keys ("west", "north", "south", "east" in some undetermined order), which is not what you want I guess

Reply:
I copy-pasted that from load_collection. I guess we can also just allow float here? Otherwise, same order as in load_collection. So if there's an easy way to handle that as in load_collection, we could also just do that. But I was a bit confused about how load_collection did it...

Reply:
in `load_collection` and `filter_bbox` we want a bbox dictionary in the end, 
but if I understand `/collections/{}/items` correctly, bbox should be an array of length `2*n`, which is totally different

Reply:
Indeed. I committed a new version (list instead dict + removing str + document the order)

Reply:
deserves some unit tests too, I'd say :)

Reply:
https://github.com/Open-EO/openeo-python-client/issues/186